docs/dotnet/dotnet-signalr-metrics.md

```diff
@@ -8,60 +8,109 @@ This document defines semantic conventions for SignalR metrics emitted by .NET c
+| `signalr.http.server.connection.duration` | Histogram | `s` | The duration of the HTTP connections on the server. [1] |
 
-**[1]:** **TODO: why http_transport? can we report protocol name as dimension and not a metric name?**
-**TODO: we should do signalr.server**
+**[1]:** Only reported for HTTP SignalR transport.
 Corresponding `EventCounter` name is `connections-duration`; Meter name is `Microsoft.AspNetCore.Http.Connections`
 <!-- endsemconv -->
 
-<!-- semconv metric.signalr.server.connection.duration(full) -->
+<!-- semconv metric.signalr.http.server.connection.duration(full) -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
-| `signalr.http_transport.status` | string | TODO [1] | `TODO` | Recommended |
-| `signalr.http_transport.transport` | string | TODO [2] | `TODO` | Recommended |
+| `signalr.transport` | string | SignalR transport - https://github.com/dotnet/aspnetcore/blob/main/src/SignalR/docs/specs/TransportProtocols.md [1] | `ServerSentEvents` | Conditionally Required: if HTTP SignalR transport is used |
+| `dotnet.error.code` | string | General-purpose error code reported by .NET, as a starter it supports terminal statuses of .NET task https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskstatus. | `Canceled`; `RanToCompletion` | Recommended |
+| `http.response.status_code` | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` | Conditionally Required: If and only if one was received/sent. |
+| [`network.protocol.name`](../general/attributes.md) | string | [OSI Application Layer](https://osi-model.com/application-layer/) or non-OSI equivalent. The value SHOULD be normalized to lowercase. | `http`; `websockets` | Recommended: if not default (`http`) |
+| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [2] | `3.1.1` | Recommended |
+| [`network.transport`](../general/attributes.md) | string | [OSI Transport Layer](https://osi-model.com/transport-layer/) or [Inter-process Communication method](https://en.wikipedia.org/wiki/Inter-process_communication). The value SHOULD be normalized to lowercase. | `tcp`; `udp` | Recommended |
 
-**[1]:** **TODO: why not signalr.status? What's HTTP-specific about it**
+**[1]:** websockets are recoded in `network.protocol.name` attribute and `signalr.transport` is not set.
 
-**[2]:** **TODO: why not network.transport?**
+**[2]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+
+`signalr.transport` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
```

Comment:
It looks like you split the enumeration and put SSE and LongPolling in one place, but websockets in a different place. That feels like it just makes the metric harder to use. In the domain of SignalR, this is a single three-valued choice of protocol and I think users would expect to find all the three values in the same enumeration.

Same issue in other signalR metrics.

Reply:
updated #1 to have all 3 protocols in `signalr.transport` under the assumption it's both client and server concept. PTAL

Reply:
Putting them all together looks much better now 👍